zcash_client_backend/src/fees/common.rs

```diff
@@ -358,10 +433,209 @@ where
                     }
                 }
             } else {
-                simple_case(change_memo)
+                simple_case()
             }
         }
     };
+    #[cfg(feature = "transparent-inputs")]
+    change.extend(
+        ephemeral_parameters
+            .ephemeral_output_amount()
+            .map(ChangeValue::ephemeral_transparent),
+    );
 
     TransactionBalance::new(change, fee).map_err(|_| overflow())
 }
+
+/// Returns a `[ChangeStrategy::DustInputs]` error if some of the inputs provided
+/// to the transaction have value less than the marginal fee, and could not be
+/// determined to have any economic value in the context of this input selection.
+///
+/// This determination is potentially conservative in the sense that outputs
+/// with value less than the marginal fee might be excluded, even though in
+/// practice they would not cause the fee to increase. Outputs with value
+/// greater than the marginal fee will never be excluded.
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn check_for_uneconomic_inputs<NoteRefT: Clone, E>(
+    transparent_inputs: &[impl transparent::InputView],
+    transparent_outputs: &[impl transparent::OutputView],
+    sapling: &impl sapling_fees::BundleView<NoteRefT>,
+    #[cfg(feature = "orchard")] orchard: &impl orchard_fees::BundleView<NoteRefT>,
+    marginal_fee: NonNegativeAmount,
+    grace_actions: usize,
+    possible_change: &[(usize, usize, usize)],
```

Comment:
I think I need the `possible_change` argument documented - is this intended to be the number of change outputs in each pool? It might be better to use a struct than a tuple here. The specifics of change will become particularly relevant when ZSAs are introduced.

Reply:
It's not a public API, so this was just the most convenient representation for passing the info from `single_change_output_balance` to `check_for_uneconomic_inputs`. We can change it at any time. But yes I will document it.